component/azstorage/azstorage.go

```diff
@@ -347,7 +347,7 @@ func (az *AzStorage) ReadInBuffer(options internal.ReadInBufferOptions) (len int
 		log.Err("AzStorage::ReadInBuffer : Failed to read %s (%s)", options.Handle.Path, err.Error())
 	}
 
-	len = int(dataLen)
+	length = int(dataLen)
```

Comment:
datalen is populated based on the slice len, is there any possibility that the read call came back with less data then requested, in that case sending back datalen will result in data corruption. We need to check if READ api in sdk returns back any sort of length of bytes actually read or the length of the slice needs to be recalculated post the read

Reply:
Rechecked this as well, the sdk API returns only error and not the length of data. In that case better to take len(options.Data) here as well instead of using the static length ?